gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenFragment.java

```diff
@@ -511,7 +524,18 @@ public void showPolygonInfoDialog() {
     AlertDialog alertDialog = builder.create();
     getStartedBtn.setOnClickListener(
         v -> {
-          alertDialog.dismiss();
+          viewModel
+              .getActiveProject()
+              .ifPresentOrElse(
+                  project -> {
+                    alertDialog.dismiss();
+                    mapContainerViewModel.setSelectedProject(Optional.of(project));
+                    mapContainerViewModel.setSelectedLayer(Optional.of(layer));
```

Comment:
What does it mean for the Map to have a layer that's "selected"? Really it sounds like the drawing tools should have their own ViewModel and View. Is it possible to refactor so that all polygon drawing functionality and state is in a separate ViewModel, and all view logic is in custom layout class?

Reply:
+1

Reply:
that will create confusion as mostly functions are used from MapContainerViewModel which is already present also the features drawing is embedded into MapContainerViewModel over there as well.

Reply:
I tried exploring a similar refactor for the feature reposition.
https://github.com/google/ground-android/pull/970

@gino-m  Is this similar to what you had it mind?

Reply:
Yes exactly, thanks for the draft proposal! Would you be able to work together on applying the same changes to this PR?

Reply:
Sure.

Reply:
https://github.com/google/ground-android/pull/970 has been merged. We should now start separating the polygon drawing code in this PR. Merging this PR as is and then refactoring seems risky as we don't have sufficient test coverage to prevent regressions.

@gino-m @HemanParbhakar Thoughts?

Reply:
ok so we cannot merge this PR until the polygon drawing code is not separated.

Reply:
That's my personal opinion. I'll let you and Gino take a final call regarding what should be done next.

Reply:
moved to PolygonDrawingViewModel.